### PR TITLE
Add ingredient 2

### PR DESCRIPTION
**Pull request by chrisspang89**
add function for new ingredient

### CODE REVIEW

File: components/IngredientDetails.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+    if (id) {
+      const storedIngredients =
+        JSON.parse(localStorage.getItem("ingredients")) || [];
+      const allIngredients = [...storedIngredients, ...ingredientsData];
+      const foundIngredient = allIngredients.find(
+        (ingredient) => ingredient._id === id
+      );
+      setIngredient(foundIngredient);
+    }
+  }, [id]);
```
Comment (NikRoe) [inline components/IngredientDetails.js]: your User Story does not mention the usage of localStorage, your Pull Request should therefore not contain localStorage.

If you wish to implement localStorage at a later point, create a User Story containing this.

Also: There was a session on localStorage mentioning a certain hook that you can use.
When you get to the point to implement localStorage, which is not in this User Story, you should use the recommended hook

File: components/IngredientsList.js [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
+      ).values(),
+    ];
+    setIngredients(allIngredients);
+  }, []);
+
+  const addIngredient = (newIngredient) => {
+    const updatedIngredients = [newIngredient, ...ingredients];
+    setIngredients(updatedIngredients);
+    localStorage.setItem("ingredients", JSON.stringify(updatedIngredients));
+  };
```
Comment (NikRoe) [inline components/IngredientsList.js]: please remove localStorage also from these lines of code.

File: components/NewIngredientForm.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      imgUrl: imgUrl || placeholderImageUrl,
+    };
+    onAddIngredient(newIngredient);
+    setName("");
+    setFlavor("");
+    setImgUrl("");
+  };
+
+  return (
+    <Form onSubmit={handleSubmit}>
```
Comment (NikRoe) [inline components/NewIngredientForm.js:26]: all of your form elements are missing their <label> please add labels and connect them with the corresponding form element

File: components/NewIngredientForm.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+const placeholderImageUrl =
+  "https://static.vecteezy.com/system/resources/previews/003/170/825/original/isolated-food-plate-fork-and-spoon-design-free-vector.jpg";
+
+const NewIngredientForm = ({ onAddIngredient, flavors }) => {
+  const [name, setName] = useState("");
+  const [flavor, setFlavor] = useState("");
+  const [imgUrl, setImgUrl] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
```
Comment (NikRoe) [inline components/NewIngredientForm.js]: please be explicit when naming arguments, parameters, functions and variables:

```suggestion
  const handleSubmit = (event) => {
    event.preventDefault();
```

File: components/NewIngredientForm.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    setImgUrl("");
+  };
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <Input
+        type="text"
+        placeholder="Name"
+        value={name}
+        onChange={(e) => setName(e.target.value)}
```
Comment (NikRoe) [inline components/NewIngredientForm.js]: please be explicit when naming arguments, parameters, functions and variables:

```suggestion
        onChange={(event) => setName(event.target.value)}
```

File: components/NewIngredientForm.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        required
+      />
+      <Select
+        value={flavor}
+        onChange={(e) => setFlavor(e.target.value)}
```
Comment (NikRoe) [inline components/NewIngredientForm.js]: please be explicit when naming arguments, parameters, functions and variables:

```suggestion
        onChange={(event) => setFlavor(event.target.value)}
```

File: components/NewIngredientForm.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+          <option key={flavor} value={flavor}>
+            {flavor}
+          </option>
+        ))}
+      </Select>
+      <Input
+        type="text"
+        placeholder="Image URL"
+        value={imgUrl}
+        onChange={(e) => setImgUrl(e.target.value)}
```
Comment (NikRoe) [inline components/NewIngredientForm.js]: please be explicit when naming arguments, parameters, functions and variables:

```suggestion
        onChange={(event) => setImgUrl(event.target.value)}
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

nice start.

When implementing a User Story, your job is only to implement what is mentioned in the User Story. Nothing less and nothing more.
The US to this Pull Request does not mention localStorage, therefore localStorage should not be part of the PR. Please remove it.

Your US also mentions that you can reach the form by clicking on a "dropdown menu button". It is currently always displayed however.

Please ensure that you truly implement what is written in the US.

Two more notes:

Elements within a form (inputs, selects etc.) should have a label that is connected to them. 

Even when getting some inspiration from chatGPT, please keep in mind what you learned during the bootcamp: Be explicit when naming things.

Keep it up!

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

this PR still contains the localStorage ;-) 

Please remove it

File: components/IngredientDetails.js [hunk 8]
```diff
@@ -1,12 +1,27 @@
...
+
+  useEffect(() => {
+    if (id) {
+      const allIngredients = ingredientsData;
+      const foundIngredient = allIngredients.find(
+        (ingredient) => ingredient._id === id
+      );
+      setIngredient(foundIngredient);
+    }
+  }, [id]);
```
Comment (NikRoe) [inline components/IngredientDetails.js]: There is no need for a useEffect or useState here. The way you had it prior to this PR was already ideal:
```suggestion
 const ingredient = ingredients.find((ingredient) => ingredientsData._id === id);
```

File: components/IngredientsList.js [hunk 9]
```diff
@@ -1,13 +1,36 @@
...
+    ];
+    setIngredients(allIngredients);
+  }, []);
+
+  const addIngredient = (newIngredient) => {
+    newIngredient._id = uuidv4(); // Assign a unique ID
+    const updatedIngredients = [newIngredient, ...ingredients];
+    setIngredients(updatedIngredients);
+    ingredientsData.push(newIngredient); // Update the ingredientsData array
+  };
```
Comment (NikRoe) [inline components/IngredientsList.js]: It is not a good pattern to modify a parameter inside a function. And pushing to the ingredientsData does not really make any sense:

```suggestion
  const addIngredient = (newIngredient) => {
    const updatedIngredients = [{_id: uuidv4(), ...newIngredient}, ...ingredients];
    setIngredients(updatedIngredients);
  };
```

File: components/IngredientsList.js [hunk 10]
```diff
@@ -1,13 +1,36 @@
...
+];
 
 const IngredientsList = () => {
+  const [ingredients, setIngredients] = useState([]);
+  useEffect(() => {
+    const allIngredients = [
+      ...new Map(ingredientsData.map((item) => [item._id, item])).values(),
+    ];
+    setIngredients(allIngredients);
+  }, []);
```
Comment (NikRoe) [inline components/IngredientsList.js]: Why do you need to do this?
Would this not work exactly the same way?

```suggestion
  const [ingredients, setIngredients] = useState(ingredientsData);
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

you made some changes that changes code from the previous PR and made it less readable.

I left you two more suggestions for more readability on top of that

File: components/IngredientsList.js
Comment (ndrbn0) [inline components/IngredientsList.js:1]: Hi @NikRoe , unfortunately in this way, some of the built features do not work. Could you please join us ?

**Review by NikRoe — APPROVED** [review]
LGTM

**Pull request merged**